### ISSUE TITLE
Semantics of optional<TimeRange> at MediaReference

### ISSUE THREAD

**Issue by palemieux** [PixarAnimationStudios/OpenTimelineIO#729]
`MediaReference::available_range` is optional according to the class definition.

`Clip::available_range()` however:

* reports an error when `MediaReference::available_range` is absent; and
* simultaneously returns a valid `TimeRange()`.

Couple of questions come to mind:

- why isn't `MediaReference::available_range` mandatory with a default value of `TimeRange()`? As a corollary, what is the difference in semantics between an absent `MediaReference::available_range` and `MediaReference::available_range` set to `TimeRange()`?
- why does `Clip::available_range()` both signal an error and return a valid value? In practice, this means that the error will be ignored, or perhaps this is merely a warning instead of an error?

**Comment by jminor** [comment]
Yes, `MediaReference::available_range` is optional, since in many cases it is not known, or expensive to discover (e.g. MissingReference, or an un-fetched URL).

`Clip::available_range()` should just pass through to it's media_reference, so it would make more sense if it also returned `optional<TimeRange>` instead of the dual error & `TimeRange()` value as it is currently implemented.
